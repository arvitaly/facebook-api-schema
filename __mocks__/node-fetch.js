"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const express = require("express");
const freeport_es6_1 = require("freeport-es6");
const fetch = require.requireActual("node-fetch").default;
const URLLib = require("url");
let port;
let server;
exports.start = jest.fn().mockImplementation(() => __awaiter(this, void 0, void 0, function* () {
    port = yield freeport_es6_1.default();
    const app = express();
    app.use(express.static(__dirname + "/../__fixtures__/urls"));
    return new Promise((resolve, reject) => {
        server = app.listen(port, (err) => {
            if (err) {
                reject(err);
                return;
            }
            resolve();
        });
    });
}));
exports.stop = jest.fn().mockImplementation(() => __awaiter(this, void 0, void 0, function* () {
    return new Promise((resolve, reject) => {
        server.close((err) => {
            if (err) {
                reject(err);
                return;
            }
            resolve();
        });
    });
}));
exports.default = jest.fn().mockImplementation((url, params) => {
    const u = URLLib.parse(url);
    u.host = `127.0.0.1:${port}`;
    u.protocol = "http";
    u.pathname = u.pathname + "/index.html";
    const newUrl = URLLib.format(u);
    return fetch(newUrl, params);
});
