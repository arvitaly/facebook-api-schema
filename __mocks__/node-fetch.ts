import express = require("express");
import freeport from "freeport-es6";
import { Server } from "http";
const fetch = require.requireActual("node-fetch").default;
import URLLib = require("url");
let port: number;
let server: Server;
export const start = jest.fn().mockImplementation(async () => {
    port = await freeport();
    const app = express();
    app.use(express.static(__dirname + "/../__fixtures__/urls"));
    return new Promise((resolve, reject) => {
        server = app.listen(port, (err: any) => {
            if (err) {
                reject(err);
                return;
            }
            resolve();
        });
    });
});
export const stop = jest.fn().mockImplementation(async () => {
    return new Promise((resolve, reject) => {
        server.close((err: any) => {
            if (err) {
                reject(err);
                return;
            }
            resolve();
        });
    });
});
export default jest.fn().mockImplementation((url: string, params: any) => {
    const u = URLLib.parse(url);
    u.host = `127.0.0.1:${port}`;
    u.protocol = "http";
    u.pathname = u.pathname + "/index.html";
    const newUrl = URLLib.format(u);
    return fetch(newUrl, params);
});
