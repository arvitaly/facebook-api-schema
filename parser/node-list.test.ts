jest.mock("node-fetch");
import { start, stop } from "./../__mocks__/node-fetch";
beforeAll(async () => start());
afterAll(async () => stop());
import getList, { Grab as GrabNodeList } from "./node-list";
it("grab from real html", async () => {
    expect(
        await GrabNodeList({})(),
    ).toMatchSnapshot();
});
it("getList", async () => {
    expect(await getList()).toMatchSnapshot();
});
