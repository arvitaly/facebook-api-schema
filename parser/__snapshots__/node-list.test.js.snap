// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getList 1`] = `
Array [
  Object {
    "description": "Represents a person gaining a game achievement",
    "link": "https://developers.facebook.com/docs/graph-api/reference/achievement",
    "name": "achievement",
    "title": "Achievement",
  },
  Object {
    "description": "A games achievement type",
    "link": "https://developers.facebook.com/docs/graph-api/reference/achievement_type",
    "name": "achievement_type",
    "title": "Achievement Type",
  },
  Object {
    "description": "A photo album",
    "link": "https://developers.facebook.com/docs/graph-api/reference/album",
    "name": "album",
    "title": "Album",
  },
  Object {
    "description": "Analytics app events export job",
    "link": "https://developers.facebook.com/docs/graph-api/reference/analytics-app-events-export",
    "name": "analytics-app-events-export",
    "title": "Analytics App Events Export",
  },
  Object {
    "description": "An app link host",
    "link": "https://developers.facebook.com/docs/graph-api/reference/app-link-host",
    "name": "app-link-host",
    "title": "App Link Host",
  },
  Object {
    "description": "An individual app request received by someone, sent by an app or another person",
    "link": "https://developers.facebook.com/docs/graph-api/reference/app-request",
    "name": "app-request",
    "title": "App Request",
  },
  Object {
    "description": "A Facebook app",
    "link": "https://developers.facebook.com/docs/graph-api/reference/application",
    "name": "application",
    "title": "Application",
  },
  Object {
    "description": "Provides access to available social context edges for this app",
    "link": "https://developers.facebook.com/docs/graph-api/reference/application-context",
    "name": "application-context",
    "title": "Application Context",
  },
  Object {
    "description": "Represents an async job request to Graph API.",
    "link": "https://developers.facebook.com/docs/graph-api/reference/async-session",
    "name": "async-session",
    "title": "Async Session",
  },
  Object {
    "description": "Definition of a rule",
    "link": "https://developers.facebook.com/docs/graph-api/reference/audience-insights-rule",
    "name": "audience-insights-rule",
    "title": "Audience Insights Rule",
  },
  Object {
    "description": "A canvas object",
    "link": "https://developers.facebook.com/docs/graph-api/reference/canvas",
    "name": "canvas",
    "title": "Canvas",
  },
  Object {
    "description": "A button inside the canvas",
    "link": "https://developers.facebook.com/docs/graph-api/reference/canvas-button",
    "name": "canvas-button",
    "title": "Canvas Button",
  },
  Object {
    "description": "A carousel inside a canvas",
    "link": "https://developers.facebook.com/docs/graph-api/reference/canvas-carousel",
    "name": "canvas-carousel",
    "title": "Canvas Carousel",
  },
  Object {
    "description": "A footer of the canvas",
    "link": "https://developers.facebook.com/docs/graph-api/reference/canvas-footer",
    "name": "canvas-footer",
    "title": "Canvas Footer",
  },
  Object {
    "description": "A header inside the canvas",
    "link": "https://developers.facebook.com/docs/graph-api/reference/canvas-header",
    "name": "canvas-header",
    "title": "Canvas Header",
  },
  Object {
    "description": "A photo inside a canvas",
    "link": "https://developers.facebook.com/docs/graph-api/reference/canvas-photo",
    "name": "canvas-photo",
    "title": "Canvas Photo",
  },
  Object {
    "description": "A product set inside the canvas",
    "link": "https://developers.facebook.com/docs/graph-api/reference/canvas-product-set",
    "name": "canvas-product-set",
    "title": "Canvas Product Set",
  },
  Object {
    "description": "A store locator inside the canvas",
    "link": "https://developers.facebook.com/docs/graph-api/reference/canvas-store-locator",
    "name": "canvas-store-locator",
    "title": "Canvas Store Locator",
  },
  Object {
    "description": "Text element of the canvas",
    "link": "https://developers.facebook.com/docs/graph-api/reference/canvas-text",
    "name": "canvas-text",
    "title": "Canvas Text",
  },
  Object {
    "description": "A video inside canvas",
    "link": "https://developers.facebook.com/docs/graph-api/reference/canvas-video",
    "name": "canvas-video",
    "title": "Canvas Video",
  },
  Object {
    "description": "A single comment",
    "link": "https://developers.facebook.com/docs/graph-api/reference/comment",
    "name": "comment",
    "title": "Comment",
  },
  Object {
    "description": "A Facebook Messages conversation between a person and a Facebook Page",
    "link": "https://developers.facebook.com/docs/graph-api/reference/conversation",
    "name": "conversation",
    "title": "Conversation",
  },
  Object {
    "description": "Debug token endpoint docs",
    "link": "https://developers.facebook.com/docs/graph-api/reference/debug_token",
    "name": "debug_token",
    "title": "Debug Token",
  },
  Object {
    "description": "A Document",
    "link": "https://developers.facebook.com/docs/graph-api/reference/doc",
    "name": "doc",
    "title": "Doc",
  },
  Object {
    "description": "A web domain claimed within Facebook Insights",
    "link": "https://developers.facebook.com/docs/graph-api/reference/domain",
    "name": "domain",
    "title": "Domain",
  },
  Object {
    "description": "The person's education history",
    "link": "https://developers.facebook.com/docs/graph-api/reference/education-experience",
    "name": "education-experience",
    "title": "Education Experience",
  },
  Object {
    "description": "An event",
    "link": "https://developers.facebook.com/docs/graph-api/reference/event",
    "name": "event",
    "title": "Event",
  },
  Object {
    "description": "This represents a user's friend list on Facebook",
    "link": "https://developers.facebook.com/docs/graph-api/reference/friend-list",
    "name": "friend-list",
    "title": "Friend List",
  },
  Object {
    "description": "A Facebook group",
    "link": "https://developers.facebook.com/docs/graph-api/reference/group",
    "name": "group",
    "title": "Group",
  },
  Object {
    "description": "A document in a Facebook group",
    "link": "https://developers.facebook.com/docs/graph-api/reference/groupdoc",
    "name": "groupdoc",
    "title": "Group Doc",
  },
  Object {
    "description": "An Instagram carousel",
    "link": "https://developers.facebook.com/docs/graph-api/reference/instagram-carousel",
    "name": "instagram-carousel",
    "title": "Instagram Carousel",
  },
  Object {
    "description": "An Instagram comment",
    "link": "https://developers.facebook.com/docs/graph-api/reference/instagram-comment",
    "name": "instagram-comment",
    "title": "Instagram Comment",
  },
  Object {
    "description": "An Instagram media",
    "link": "https://developers.facebook.com/docs/graph-api/reference/instagram-media",
    "name": "instagram-media",
    "title": "Instagram Media",
  },
  Object {
    "description": "An Instagram user",
    "link": "https://developers.facebook.com/docs/graph-api/reference/instagram-user",
    "name": "instagram-user",
    "title": "Instagram User",
  },
  Object {
    "description": "Page milestone information",
    "link": "https://developers.facebook.com/docs/graph-api/reference/life-event",
    "name": "life-event",
    "title": "Life Event",
  },
  Object {
    "description": "A link shared on Facebook",
    "link": "https://developers.facebook.com/docs/graph-api/reference/link",
    "name": "link",
    "title": "Link",
  },
  Object {
    "description": "An EntLiveEncoder is for the live encoders that can be associated withvideo broadcasts. This is part of the reference live encoder API.",
    "link": "https://developers.facebook.com/docs/graph-api/reference/live-encoder",
    "name": "live-encoder",
    "title": "Live Encoder",
  },
  Object {
    "description": "A live video",
    "link": "https://developers.facebook.com/docs/graph-api/reference/live-video",
    "name": "live-video",
    "title": "Live Video",
  },
  Object {
    "description": "A mailing address object",
    "link": "https://developers.facebook.com/docs/graph-api/reference/mailing-address",
    "name": "mailing-address",
    "title": "Mailing Address",
  },
  Object {
    "description": "Media fingerprint",
    "link": "https://developers.facebook.com/docs/graph-api/reference/media-fingerprint",
    "name": "media-fingerprint",
    "title": "Media Fingerprint",
  },
  Object {
    "description": "A single message in Facebook Messenger",
    "link": "https://developers.facebook.com/docs/graph-api/reference/message",
    "name": "message",
    "title": "Message",
  },
  Object {
    "description": "A milestone for a Facebook Page",
    "link": "https://developers.facebook.com/docs/graph-api/reference/milestone",
    "name": "milestone",
    "title": "Milestone",
  },
  Object {
    "description": "A native offer</code> represents an Offer on Facebook.  The <code>/{offer_id}</code> node returns a single <code>native offer</code>.  Each <code>native offer</code> requires a <code>view to be rendered to users.",
    "link": "https://developers.facebook.com/docs/graph-api/reference/native-offer",
    "name": "native-offer",
    "title": "Native Offer",
  },
  Object {
    "description": "An individual unread Facebook notification",
    "link": "https://developers.facebook.com/docs/graph-api/reference/notification",
    "name": "notification",
    "title": "Notification",
  },
  Object {
    "description": "Comments on an object",
    "link": "https://developers.facebook.com/docs/graph-api/reference/object/comments",
    "name": "comments",
    "title": "Object Comments",
  },
  Object {
    "description": "Likes for an object",
    "link": "https://developers.facebook.com/docs/graph-api/reference/object/likes",
    "name": "likes",
    "title": "Object Likes",
  },
  Object {
    "description": "Shares of this object",
    "link": "https://developers.facebook.com/docs/graph-api/reference/object/sharedposts",
    "name": "sharedposts",
    "title": "Object Sharedposts",
  },
  Object {
    "description": "An Open Graph action type",
    "link": "https://developers.facebook.com/docs/graph-api/reference/open-graph-action-type",
    "name": "open-graph-action-type",
    "title": "Open Graph Action Type",
  },
  Object {
    "description": "Social context",
    "link": "https://developers.facebook.com/docs/graph-api/reference/open-graph-context",
    "name": "open-graph-context",
    "title": "Open Graph Context",
  },
  Object {
    "description": "An Open Graph object type",
    "link": "https://developers.facebook.com/docs/graph-api/reference/open-graph-object-type",
    "name": "open-graph-object-type",
    "title": "Open Graph Object Type",
  },
  Object {
    "description": "A Facebook page",
    "link": "https://developers.facebook.com/docs/graph-api/reference/page",
    "name": "page",
    "title": "Page",
  },
  Object {
    "description": "Page's admin note",
    "link": "https://developers.facebook.com/docs/graph-api/reference/page-admin-note",
    "name": "page-admin-note",
    "title": "Page Admin Note",
  },
  Object {
    "description": "Page's call-to-action",
    "link": "https://developers.facebook.com/docs/graph-api/reference/page-call-to-action",
    "name": "page-call-to-action",
    "title": "Page Call To Action",
  },
  Object {
    "description": "Page's label",
    "link": "https://developers.facebook.com/docs/graph-api/reference/page-label",
    "name": "page-label",
    "title": "Page Label",
  },
  Object {
    "description": "Insights and metrics for an object",
    "link": "https://developers.facebook.com/docs/graph-api/reference/insights",
    "name": "insights",
    "title": "Page/insights",
  },
  Object {
    "description": "A node to get Native Component workflows info.",
    "link": "https://developers.facebook.com/docs/graph-api/reference/pages-platform-component-flow-service-config",
    "name": "pages-platform-component-flow-service-config",
    "title": "Pages Platform Component Flow Service Config",
  },
  Object {
    "description": "A single payment",
    "link": "https://developers.facebook.com/docs/graph-api/reference/payment",
    "name": "payment",
    "title": "Payment",
  },
  Object {
    "description": "This represents a Photo on Facebook",
    "link": "https://developers.facebook.com/docs/graph-api/reference/photo",
    "name": "photo",
    "title": "Photo",
  },
  Object {
    "description": "A place",
    "link": "https://developers.facebook.com/docs/graph-api/reference/place",
    "name": "place",
    "title": "Place",
  },
  Object {
    "description": "A Place Tag",
    "link": "https://developers.facebook.com/docs/graph-api/reference/place-tag",
    "name": "place-tag",
    "title": "Place Tag",
  },
  Object {
    "description": "The category of a place Page",
    "link": "https://developers.facebook.com/docs/graph-api/reference/place-topic",
    "name": "place-topic",
    "title": "Place Topic",
  },
  Object {
    "description": "Special syntax for the link field.",
    "link": "https://developers.facebook.com/docs/graph-api/reference/post",
    "name": "post",
    "title": "Post",
  },
  Object {
    "description": "A profile, which can be a user, page, group or event",
    "link": "https://developers.facebook.com/docs/graph-api/reference/profile",
    "name": "profile",
    "title": "Profile",
  },
  Object {
    "description": "A promotional info for the post.",
    "link": "https://developers.facebook.com/docs/graph-api/reference/promotion-info",
    "name": "promotion-info",
    "title": "Promotion Info",
  },
  Object {
    "description": "A dynamically generated Post, used for Dynamic Ad Creatives",
    "link": "https://developers.facebook.com/docs/graph-api/reference/rtb-dynamic-post",
    "name": "rtb-dynamic-post",
    "title": "RTBDynamic Post",
  },
  Object {
    "description": "An app request received by a person.  A request can be sent by an app or another person",
    "link": "https://developers.facebook.com/docs/graph-api/reference/request",
    "name": "request",
    "title": "Request",
  },
  Object {
    "description": "Entity to describe an offer. One needs to be whitelisted to use this API. Please contact the support group and ask them to reach out to the team that is responsible for this API to request to be whitelisted.",
    "link": "https://developers.facebook.com/docs/graph-api/reference/sales-promo",
    "name": "sales-promo",
    "title": "Sales Promo",
  },
  Object {
    "description": "A saved message response for a Facebook Page",
    "link": "https://developers.facebook.com/docs/graph-api/reference/saved-message-response",
    "name": "saved-message-response",
    "title": "Saved Message Response",
  },
  Object {
    "description": "A status message in a profile's feed",
    "link": "https://developers.facebook.com/docs/graph-api/reference/status",
    "name": "status",
    "title": "Status",
  },
  Object {
    "description": "A test user associated with a Facebook app. Test users are created and associated using the /app/accounts/test-users edge or the App Dashboard.",
    "link": "https://developers.facebook.com/docs/graph-api/reference/test-user",
    "name": "test-user",
    "title": "Test User",
  },
  Object {
    "description": "A messages thread in Facebook Messenger",
    "link": "https://developers.facebook.com/docs/graph-api/reference/thread",
    "name": "thread",
    "title": "Thread",
  },
  Object {
    "description": "Shares, app links, and Open Graph objects for a URL",
    "link": "https://developers.facebook.com/docs/graph-api/reference/url",
    "name": "url",
    "title": "URL",
  },
  Object {
    "description": "Returns a single user node",
    "link": "https://developers.facebook.com/docs/graph-api/reference/user",
    "name": "user",
    "title": "User",
  },
  Object {
    "description": "Social context for a person",
    "link": "https://developers.facebook.com/docs/graph-api/reference/user-context",
    "name": "user-context",
    "title": "User Context",
  },
  Object {
    "description": "A Video",
    "link": "https://developers.facebook.com/docs/graph-api/reference/video",
    "name": "video",
    "title": "Video",
  },
  Object {
    "description": "A video copyright",
    "link": "https://developers.facebook.com/docs/graph-api/reference/video-copyright",
    "name": "video-copyright",
    "title": "Video Copyright",
  },
  Object {
    "description": "A video copyright rules",
    "link": "https://developers.facebook.com/docs/graph-api/reference/video-copyright-rule",
    "name": "video-copyright-rule",
    "title": "Video Copyright Rule",
  },
  Object {
    "description": "A playlist for videos",
    "link": "https://developers.facebook.com/docs/graph-api/reference/video-list",
    "name": "video-list",
    "title": "Video List",
  },
  Object {
    "description": "Information about a user's work",
    "link": "https://developers.facebook.com/docs/graph-api/reference/work-experience",
    "name": "work-experience",
    "title": "Work Experience",
  },
]
`;

exports[`grab from real html 1`] = `
Array [
  Object {
    "description": "Represents a person gaining a game achievement",
    "name": Object {
      "href": "/docs/graph-api/reference/achievement",
      "title": "Achievement",
    },
  },
  Object {
    "description": "A games achievement type",
    "name": Object {
      "href": "/docs/graph-api/reference/achievement_type",
      "title": "Achievement Type",
    },
  },
  Object {
    "description": "A photo album",
    "name": Object {
      "href": "/docs/graph-api/reference/album",
      "title": "Album",
    },
  },
  Object {
    "description": "Analytics app events export job",
    "name": Object {
      "href": "/docs/graph-api/reference/analytics-app-events-export",
      "title": "Analytics App Events Export",
    },
  },
  Object {
    "description": "An app link host",
    "name": Object {
      "href": "/docs/graph-api/reference/app-link-host",
      "title": "App Link Host",
    },
  },
  Object {
    "description": "An individual app request received by someone, sent by an app or another person",
    "name": Object {
      "href": "/docs/graph-api/reference/app-request",
      "title": "App Request",
    },
  },
  Object {
    "description": "A Facebook app",
    "name": Object {
      "href": "/docs/graph-api/reference/application",
      "title": "Application",
    },
  },
  Object {
    "description": "Provides access to available social context edges for this app",
    "name": Object {
      "href": "/docs/graph-api/reference/application-context",
      "title": "Application Context",
    },
  },
  Object {
    "description": "Represents an async job request to Graph API.",
    "name": Object {
      "href": "/docs/graph-api/reference/async-session",
      "title": "Async Session",
    },
  },
  Object {
    "description": "Definition of a rule",
    "name": Object {
      "href": "/docs/graph-api/reference/audience-insights-rule",
      "title": "Audience Insights Rule",
    },
  },
  Object {
    "description": "A canvas object",
    "name": Object {
      "href": "/docs/graph-api/reference/canvas",
      "title": "Canvas",
    },
  },
  Object {
    "description": "A button inside the canvas",
    "name": Object {
      "href": "/docs/graph-api/reference/canvas-button",
      "title": "Canvas Button",
    },
  },
  Object {
    "description": "A carousel inside a canvas",
    "name": Object {
      "href": "/docs/graph-api/reference/canvas-carousel",
      "title": "Canvas Carousel",
    },
  },
  Object {
    "description": "A footer of the canvas",
    "name": Object {
      "href": "/docs/graph-api/reference/canvas-footer",
      "title": "Canvas Footer",
    },
  },
  Object {
    "description": "A header inside the canvas",
    "name": Object {
      "href": "/docs/graph-api/reference/canvas-header",
      "title": "Canvas Header",
    },
  },
  Object {
    "description": "A photo inside a canvas",
    "name": Object {
      "href": "/docs/graph-api/reference/canvas-photo",
      "title": "Canvas Photo",
    },
  },
  Object {
    "description": "A product set inside the canvas",
    "name": Object {
      "href": "/docs/graph-api/reference/canvas-product-set",
      "title": "Canvas Product Set",
    },
  },
  Object {
    "description": "A store locator inside the canvas",
    "name": Object {
      "href": "/docs/graph-api/reference/canvas-store-locator",
      "title": "Canvas Store Locator",
    },
  },
  Object {
    "description": "Text element of the canvas",
    "name": Object {
      "href": "/docs/graph-api/reference/canvas-text",
      "title": "Canvas Text",
    },
  },
  Object {
    "description": "A video inside canvas",
    "name": Object {
      "href": "/docs/graph-api/reference/canvas-video",
      "title": "Canvas Video",
    },
  },
  Object {
    "description": "A single comment",
    "name": Object {
      "href": "/docs/graph-api/reference/comment",
      "title": "Comment",
    },
  },
  Object {
    "description": "A Facebook Messages conversation between a person and a Facebook Page",
    "name": Object {
      "href": "/docs/graph-api/reference/conversation",
      "title": "Conversation",
    },
  },
  Object {
    "description": "Debug token endpoint docs",
    "name": Object {
      "href": "/docs/graph-api/reference/debug_token",
      "title": "Debug Token",
    },
  },
  Object {
    "description": "A Document",
    "name": Object {
      "href": "/docs/graph-api/reference/doc",
      "title": "Doc",
    },
  },
  Object {
    "description": "A web domain claimed within Facebook Insights",
    "name": Object {
      "href": "/docs/graph-api/reference/domain",
      "title": "Domain",
    },
  },
  Object {
    "description": "The person's education history",
    "name": Object {
      "href": "/docs/graph-api/reference/education-experience",
      "title": "Education Experience",
    },
  },
  Object {
    "description": "An event",
    "name": Object {
      "href": "/docs/graph-api/reference/event",
      "title": "Event",
    },
  },
  Object {
    "description": "This represents a user's friend list on Facebook",
    "name": Object {
      "href": "/docs/graph-api/reference/friend-list",
      "title": "Friend List",
    },
  },
  Object {
    "description": "A Facebook group",
    "name": Object {
      "href": "/docs/graph-api/reference/group",
      "title": "Group",
    },
  },
  Object {
    "description": "A document in a Facebook group",
    "name": Object {
      "href": "/docs/graph-api/reference/groupdoc",
      "title": "Group Doc",
    },
  },
  Object {
    "description": "An Instagram carousel",
    "name": Object {
      "href": "/docs/graph-api/reference/instagram-carousel",
      "title": "Instagram Carousel",
    },
  },
  Object {
    "description": "An Instagram comment",
    "name": Object {
      "href": "/docs/graph-api/reference/instagram-comment",
      "title": "Instagram Comment",
    },
  },
  Object {
    "description": "An Instagram media",
    "name": Object {
      "href": "/docs/graph-api/reference/instagram-media",
      "title": "Instagram Media",
    },
  },
  Object {
    "description": "An Instagram user",
    "name": Object {
      "href": "/docs/graph-api/reference/instagram-user",
      "title": "Instagram User",
    },
  },
  Object {
    "description": "Page milestone information",
    "name": Object {
      "href": "/docs/graph-api/reference/life-event",
      "title": "Life Event",
    },
  },
  Object {
    "description": "A link shared on Facebook",
    "name": Object {
      "href": "/docs/graph-api/reference/link",
      "title": "Link",
    },
  },
  Object {
    "description": "An EntLiveEncoder is for the live encoders that can be associated withvideo broadcasts. This is part of the reference live encoder API.",
    "name": Object {
      "href": "/docs/graph-api/reference/live-encoder",
      "title": "Live Encoder",
    },
  },
  Object {
    "description": "A live video",
    "name": Object {
      "href": "/docs/graph-api/reference/live-video",
      "title": "Live Video",
    },
  },
  Object {
    "description": "A mailing address object",
    "name": Object {
      "href": "/docs/graph-api/reference/mailing-address",
      "title": "Mailing Address",
    },
  },
  Object {
    "description": "Media fingerprint",
    "name": Object {
      "href": "/docs/graph-api/reference/media-fingerprint",
      "title": "Media Fingerprint",
    },
  },
  Object {
    "description": "A single message in Facebook Messenger",
    "name": Object {
      "href": "/docs/graph-api/reference/message",
      "title": "Message",
    },
  },
  Object {
    "description": "A milestone for a Facebook Page",
    "name": Object {
      "href": "/docs/graph-api/reference/milestone",
      "title": "Milestone",
    },
  },
  Object {
    "description": "A <code>native offer</code> represents an Offer on Facebook.  The <code>/{offer_id}</code> node returns a single <code>native offer</code>.  Each <code>native offer</code> requires a <code>view</code> to be rendered to users.",
    "name": Object {
      "href": "/docs/graph-api/reference/native-offer",
      "title": "Native Offer",
    },
  },
  Object {
    "description": "An individual unread Facebook notification",
    "name": Object {
      "href": "/docs/graph-api/reference/notification",
      "title": "Notification",
    },
  },
  Object {
    "description": "Comments on an object",
    "name": Object {
      "href": "/docs/graph-api/reference/object/comments",
      "title": "Object Comments",
    },
  },
  Object {
    "description": "Likes for an object",
    "name": Object {
      "href": "/docs/graph-api/reference/object/likes",
      "title": "Object Likes",
    },
  },
  Object {
    "description": "Shares of this object",
    "name": Object {
      "href": "/docs/graph-api/reference/object/sharedposts",
      "title": "Object Sharedposts",
    },
  },
  Object {
    "description": "An Open Graph action type",
    "name": Object {
      "href": "/docs/graph-api/reference/open-graph-action-type",
      "title": "Open Graph Action Type",
    },
  },
  Object {
    "description": "Social context",
    "name": Object {
      "href": "/docs/graph-api/reference/open-graph-context",
      "title": "Open Graph Context",
    },
  },
  Object {
    "description": "An Open Graph object type",
    "name": Object {
      "href": "/docs/graph-api/reference/open-graph-object-type",
      "title": "Open Graph Object Type",
    },
  },
  Object {
    "description": "A Facebook page",
    "name": Object {
      "href": "/docs/graph-api/reference/page",
      "title": "Page",
    },
  },
  Object {
    "description": "Page's admin note",
    "name": Object {
      "href": "/docs/graph-api/reference/page-admin-note",
      "title": "Page Admin Note",
    },
  },
  Object {
    "description": "Page's call-to-action",
    "name": Object {
      "href": "/docs/graph-api/reference/page-call-to-action",
      "title": "Page Call To Action",
    },
  },
  Object {
    "description": "Page's label",
    "name": Object {
      "href": "/docs/graph-api/reference/page-label",
      "title": "Page Label",
    },
  },
  Object {
    "description": "Insights and metrics for an object",
    "name": Object {
      "href": "/docs/graph-api/reference/insights",
      "title": "Page/insights",
    },
  },
  Object {
    "description": "A node to get Native Component workflows info.",
    "name": Object {
      "href": "/docs/graph-api/reference/pages-platform-component-flow-service-config",
      "title": "Pages Platform Component Flow Service Config",
    },
  },
  Object {
    "description": "A single payment",
    "name": Object {
      "href": "/docs/graph-api/reference/payment",
      "title": "Payment",
    },
  },
  Object {
    "description": "This represents a Photo on Facebook",
    "name": Object {
      "href": "/docs/graph-api/reference/photo",
      "title": "Photo",
    },
  },
  Object {
    "description": "A place",
    "name": Object {
      "href": "/docs/graph-api/reference/place",
      "title": "Place",
    },
  },
  Object {
    "description": "A Place Tag",
    "name": Object {
      "href": "/docs/graph-api/reference/place-tag",
      "title": "Place Tag",
    },
  },
  Object {
    "description": "The category of a place Page",
    "name": Object {
      "href": "/docs/graph-api/reference/place-topic",
      "title": "Place Topic",
    },
  },
  Object {
    "description": "Special syntax for the link field.",
    "name": Object {
      "href": "/docs/graph-api/reference/post",
      "title": "Post",
    },
  },
  Object {
    "description": "A profile, which can be a user, page, group or event",
    "name": Object {
      "href": "/docs/graph-api/reference/profile",
      "title": "Profile",
    },
  },
  Object {
    "description": "A promotional info for the post.",
    "name": Object {
      "href": "/docs/graph-api/reference/promotion-info",
      "title": "Promotion Info",
    },
  },
  Object {
    "description": "A dynamically generated Post, used for Dynamic Ad Creatives",
    "name": Object {
      "href": "/docs/graph-api/reference/rtb-dynamic-post",
      "title": "RTBDynamic Post",
    },
  },
  Object {
    "description": "An app request received by a person.  A request can be sent by an app or another person",
    "name": Object {
      "href": "/docs/graph-api/reference/request",
      "title": "Request",
    },
  },
  Object {
    "description": "Entity to describe an offer. One needs to be whitelisted to use this API. Please contact the support group and ask them to reach out to the team that is responsible for this API to request to be whitelisted.",
    "name": Object {
      "href": "/docs/graph-api/reference/sales-promo",
      "title": "Sales Promo",
    },
  },
  Object {
    "description": "A saved message response for a Facebook Page",
    "name": Object {
      "href": "/docs/graph-api/reference/saved-message-response",
      "title": "Saved Message Response",
    },
  },
  Object {
    "description": "A status message in a profile's feed",
    "name": Object {
      "href": "/docs/graph-api/reference/status",
      "title": "Status",
    },
  },
  Object {
    "description": "A test user associated with a Facebook app. Test users are created and associated using the /app/accounts/test-users edge or the App Dashboard.",
    "name": Object {
      "href": "/docs/graph-api/reference/test-user",
      "title": "Test User",
    },
  },
  Object {
    "description": "A messages thread in Facebook Messenger",
    "name": Object {
      "href": "/docs/graph-api/reference/thread",
      "title": "Thread",
    },
  },
  Object {
    "description": "Shares, app links, and Open Graph objects for a URL",
    "name": Object {
      "href": "/docs/graph-api/reference/url",
      "title": "URL",
    },
  },
  Object {
    "description": "Returns a single user node",
    "name": Object {
      "href": "/docs/graph-api/reference/user",
      "title": "User",
    },
  },
  Object {
    "description": "Social context for a person",
    "name": Object {
      "href": "/docs/graph-api/reference/user-context",
      "title": "User Context",
    },
  },
  Object {
    "description": "A Video",
    "name": Object {
      "href": "/docs/graph-api/reference/video",
      "title": "Video",
    },
  },
  Object {
    "description": "A video copyright",
    "name": Object {
      "href": "/docs/graph-api/reference/video-copyright",
      "title": "Video Copyright",
    },
  },
  Object {
    "description": "A video copyright rules",
    "name": Object {
      "href": "/docs/graph-api/reference/video-copyright-rule",
      "title": "Video Copyright Rule",
    },
  },
  Object {
    "description": "A playlist for videos",
    "name": Object {
      "href": "/docs/graph-api/reference/video-list",
      "title": "Video List",
    },
  },
  Object {
    "description": "Information about a user's work",
    "name": Object {
      "href": "/docs/graph-api/reference/work-experience",
      "title": "Work Experience",
    },
  },
]
`;
